packages/custom-webpack/src/custom-webpack-builder.spec.ts

```diff
@@ -20,39 +20,75 @@ const customWebpackConfig = {
+				options: {
+					prod: true
+				}
+			}
+		]
+	}
+};
+
+function createConfigFile(fileName: string) {
+	jest.mock(`${__dirname}/${fileName}`, () => customWebpackConfig, {virtual: true});
 }
 
-describe('CustomWebpackBuilder test', () => {
-	let fileName: string;
+function createConfigFunction(fileName: string) {
+	jest.mock(`${__dirname}/${fileName}`, () => customWebpackFunction, {virtual: true})
+}
 
+describe('CustomWebpackBuilder test', () => {
 	it('Should load webpack.config.js if no path specified', () => {
-		fileName = defaultWebpackConfigPath;
-		createConfigFile(fileName);
-		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {}, baseWebpackConfig);
+		createConfigFile(defaultWebpackConfigPath);
+		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {}, baseWebpackConfig, {});
 		expect(WebpackConfigMerger.merge).toHaveBeenCalledWith(baseWebpackConfig, customWebpackConfig, undefined, undefined);
 	});
 
 	it('Should load the file specified in configuration', () => {
-		fileName = 'extra-webpack.config.js';
+		const fileName = 'extra-webpack.config.js';
 		createConfigFile(fileName);
-		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {path: 'extra-webpack.config.js'}, baseWebpackConfig);
+		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {path: fileName}, baseWebpackConfig, {});
 		expect(WebpackConfigMerger.merge).toHaveBeenCalledWith(baseWebpackConfig, customWebpackConfig, undefined, undefined);
 	});
 
 	it('Should pass on merge strategies', () => {
-		fileName = defaultWebpackConfigPath;
-		createConfigFile(fileName);
+		createConfigFile(defaultWebpackConfigPath);
 		const mergeStrategies: MergeStrategies = {'blah': 'prepend'};
-		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {mergeStrategies}, baseWebpackConfig);
+		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {mergeStrategies}, baseWebpackConfig, {});
 		expect(WebpackConfigMerger.merge).toHaveBeenCalledWith(baseWebpackConfig, customWebpackConfig, mergeStrategies, undefined);
 	});
 
 	it('Should pass on replaceDuplicatePlugins flag', () => {
-		fileName = defaultWebpackConfigPath;
-		createConfigFile(fileName);
-		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {replaceDuplicatePlugins: true}, baseWebpackConfig);
+		createConfigFile(defaultWebpackConfigPath);
+		CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {replaceDuplicatePlugins: true}, baseWebpackConfig, {});
 		expect(WebpackConfigMerger.merge).toHaveBeenCalledWith(baseWebpackConfig, customWebpackConfig, undefined, true);
 	});
-});
\ No newline at end of file
+
+	it('Should execute custom function on configuration', () => {
+		const fileName = 'custom-webpack.config.js';
+		createConfigFunction(fileName);
+		const mergedConfig = CustomWebpackBuilder.buildWebpackConfig(__dirname as Path, {path: fileName}, baseWebpackConfig, {env: 'prod'});
+		 expect(mergedConfig).toEqual(customWebpackFunctionObj);
+	});
+});
```

Comment:
Would be great to add another test checking that the second parameter was actually received in the function.

Reply:
testing that `prod: options.env === 'prod'` actually returns `prod: true` is not enough ?

Reply:
anyway, how would you test it explicitely ?

Reply:
I meant that the preferred way is to have separate tests for separate functionality. Here you have a test that tests both that the function was applied and that the second parameter is received.  
Essentially if one of them breaks the test will fail and you'll have to understand which one has failed.  
The easiest way to test the "parameter was passed" thing is to pass a spy instead of a function (`jest.fn()`) and make sure that it `hasBeenCalledWith` the right parameters.

Reply:
done